openslides_backend/presenter/get_user_related_models.py

```diff
@@ -0,0 +1,159 @@
+from .presenter import register_presenter
+
+get_user_related_models_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "title": "get_user_related_models",
+        "description": "get user ids related models",
+        "properties": {
+            "user_ids": {
+                "type": "array",
+                "item": {"type": "integer"},
+            },
+        },
+        "required": ["user_ids"],
+        "additionalProperties": False,
+    }
+)
+
+
+@register_presenter("get_user_related_models")
+class GetUserRelatedModels(BasePresenter):
+    """
+    Collects related models of the user_ids.
+    """
+
+    schema = get_user_related_models_schema
+
+    def get_result(self) -> Any:
+        result: Dict[str, Any] = {}
+        for user_id in self.data["user_ids"]:
+            result[str(user_id)] = {}
+            committees_data = self.get_committees_data(user_id)
+            meetings_data = self.get_meetings_data(user_id)
+            if committees_data:
+                result[str(user_id)]["committees"] = committees_data
+            if meetings_data:
+                result[str(user_id)]["meetings"] = meetings_data
+        self.check_permissions(result)
+        return result
+
+    def check_permissions(self, result: Any) -> None:
+        """It first collects the meetings which are included and checks them."""
+        if has_organization_management_level(
+            self.datastore, self.user_id, OrganizationManagementLevel.CAN_MANAGE_USERS
+        ):
+            return
+        for user_id in result:
+            meeting_ids = []
+            for meeting in result[user_id].get("meetings", []):
+                meeting_ids.append(meeting["id"])
+            if all(
+                has_perm(
+                    self.datastore,
+                    self.user_id,
+                    Permissions.User.CAN_MANAGE,
+                    meeting_id,
+                )
+                for meeting_id in meeting_ids
+            ):
+                return
+        raise MissingPermission(OrganizationManagementLevel.CAN_MANAGE_USERS)
+
+    def get_committees_data(self, user_id: int) -> List[Dict[str, Any]]:
+        committees_data = []
+        user = self.datastore.get(
+            FullQualifiedId(Collection("user"), user_id), ["committee_ids"]
+        )
+        return_committees = False
+        if not user.get("committee_ids"):
+            return []
+        gmr = GetManyRequest(
+            Collection("committee"), user["committee_ids"], ["id", "name"]
+        )
+        committees = (
+            self.datastore.get_many([gmr]).get(Collection("committee"), {}).values()
+        )
+        for committee in committees:
+            committee_id = committee["id"]
+            user2 = self.datastore.get(
+                FullQualifiedId(Collection("user"), user_id),
+                [f"committee_${committee_id}_management_level"],
+            )
+            if (
```

Comment:
You don't need the if-block and the variable return_committees. See the changed/clarified requirements in [wiki](https://github.com/OpenSlides/OpenSlides/wiki/get_user_related_models#logic).